scripts/generate_spirv.py

```diff
@@ -179,104 +162,159 @@ def write(words, filename, apiname, outdir = None):
+                 continue
+             for filename in os.listdir(dir_path):
+                if filename.split(".")[-1].lower() in shader_type: # Use lower() for case-insensitivity
+                    shaders_to_compile.append(os.path.join(dir_path, filename))
+
+    except Exception as e:
+         sys.exit(f"Error finding shader directories: {e}")
+
+    if not shaders_to_compile:
+        sys.exit("No shader files found to compile.")
+
+    external_dir = None
+    for path_suffix in ['external/Debug/64', 'external/Release/64', 'external']:
+        try:
+            potential_dir = common_ci.RepoRelative(path_suffix)
+            if os.path.isdir(potential_dir):
+                external_dir = potential_dir
+                break
+        except Exception:
+             continue
+
+    if not external_dir:
+         print("Warning: Could not automatically determine external tools directory.", file=sys.stderr)
 
     # default glslangValidator path
-    glslang = common_ci.RepoRelative(os.path.join(external_dir, 'glslang/build/install/bin/glslang'))
-    if args.glslang:
-        glslang = args.glslang
+    glslang_default = os.path.join(external_dir, 'glslang/build/install/bin/glslang') if external_dir else 'glslang'
+    glslang = args.glslang if args.glslang else glslang_default
     if not shutil.which(glslang):
-        sys.exit("Cannot find glslangValidator " + glslang)
+        if not shutil.which('glslang'):
+             sys.exit(f"Cannot find glslangValidator executable: Tried '{glslang}' and 'glslang' in PATH.")
+        else:
+             glslang = 'glslang'
 
 
     # default spirv-opt path
-    spirv_opt =  common_ci.RepoRelative(os.path.join(external_dir, 'SPIRV-Tools/build/install/bin/spirv-opt'))
-    if args.spirv_opt:
-        spirv_opt = args.spirv_opt
+    spirv_opt_default = os.path.join(external_dir, 'SPIRV-Tools/build/install/bin/spirv-opt') if external_dir else 'spirv-opt'
+    spirv_opt = args.spirv_opt if args.spirv_opt else spirv_opt_default
     if not shutil.which(spirv_opt):
-        sys.exit("Cannot find spirv-opt " + spirv_opt)
-
-    if args.shader:
-        if not os.path.isfile(args.shader):
-            sys.exit("Cannot find infilename " + args.shader)
-        shaders_to_compile = [args.shader]
+        if not shutil.which('spirv-opt'):
+            sys.exit(f"Cannot find spirv-opt executable: Tried '{spirv_opt}' and 'spirv-opt' in PATH.")
+        else:
+            spirv_opt = 'spirv-opt'
+    
+    # compile shaders
+    compiled_shader_data = []
+    compilation_failed = False
 
     if len(shaders_to_compile) == 1 or args.single_thread:
         for shader in shaders_to_compile:
-            process_shader(shader, gpu_shaders_dir, glslang, spirv_opt, args.targetenv, args.api, args.outdir)
+            try:
+                result = process_shader(shader, gpu_shaders_dir, glslang, spirv_opt, args.targetenv)
+                compiled_shader_data.append(result)
+            except Exception as e:
```

Comment:
```suggestion
            except Exception:
```

same below, `e` is no used